### PR TITLE
Preparation for merge of GPU codepath into main branch.

### PR DESCRIPTION
**Pull request by marsdeno**
This should be cosmetic only : move original codepath to cpu directory.
Should make it obvious that subsequent GPU merge does not interact with CPU codepath

### CODE REVIEW

File: src/programs/CMakeLists.txt [hunk 1]
```diff
@@ -30,18 +30,20 @@ if( HAVE_TOOLS )
...
 endif()
 
-foreach( prec sp dp )
-  if( HAVE_${prec} )
-    ecbuild_add_executable(TARGET  ectrans-benchmark-${prec}
+if ( HAVE_CPU )
+  foreach( prec sp dp )
+    if( HAVE_${prec} )
+      ecbuild_add_executable(TARGET  ectrans-benchmark-${prec}
                            SOURCES ectrans-benchmark.F90
```
Comment (samhatfield) [inline src/programs/CMakeLists.txt:37]: Minor point: could you indent these lines a couple spaces?

**Comment by samhatfield** [comment]
Just a quick check but it looks good to me. I guess you copied the contents of src/trans/CMakeLists.txt verbatim to src/trans/cpu/CMakeLists.txt?

Let's see what the CI says!

**Comment by samhatfield** [comment]
All good to me. @wdeconinck can confirm, but shouldn't this be targeting develop?

**Comment by marsdeno** [comment]
> All good to me. @wdeconinck can confirm, but shouldn't this be targeting develop?

Duh! Absolutely right ...

**Comment by marsdeno** [comment]
Closing to redo properly over develop

**Comment by marsdeno** [comment]
@wdeconinck  you have a point, that would be a pain. I can try and rebase the gpu branch over this preparation branch, and see what happens, I guess.

**Pull request closed without merging**